backend/src/apis/gms/model/GmsUser.ts

```diff
@@ -0,0 +1,108 @@
+    this.lastName = this.getGmsLastName(user)
+    this.alias = this.getGmsAlias(user)
+    this.type = GmsLocationType.GMS_LOCATION_TYPE_RANDOM
+    this.location = null
+  }
+
+  id: number
+  userUuid: string
+  communityUuid: string
+  email: string | undefined
+  countryCode: string | undefined
+  mobile: string | undefined
+  status: number
+  createdAt: Date
+  updatedAt: Date
+  firstName: string | undefined
+  lastName: string | undefined
+  alias: string | undefined
+  type: number
+  address: string | undefined
+  city: string | undefined
+  state: string
+  country: string | undefined
+  zipCode: string | undefined
+  language: string
+  location: unknown
+
+  private getGmsAlias(user: dbUser): string | undefined {
+    if (
+      user.gmsAllowed &&
+      user.alias &&
+      user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_ALIAS_OR_INITALS
+    ) {
+      return user.alias
+    }
+  }
+
+  private getGmsFirstName(user: dbUser): string | undefined {
+    if (
+      user.gmsAllowed &&
+      (user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_FIRST ||
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_FIRST_INITIAL ||
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_FULL)
+    ) {
+      return user.firstName
+    }
+    if (
+      user.gmsAllowed &&
+      ((!user.alias &&
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_ALIAS_OR_INITALS) ||
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_INITIALS)
+    ) {
+      return user.firstName.substring(0, 1)
+    }
+  }
+
+  private getGmsLastName(user: dbUser): string | undefined {
+    if (user.gmsAllowed && user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_FULL) {
+      return user.lastName
+    }
+    if (
+      user.gmsAllowed &&
+      ((!user.alias &&
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_ALIAS_OR_INITALS) ||
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_FIRST_INITIAL ||
+        user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_INITIALS)
+    ) {
+      return user.lastName.substring(0, 1)
+    }
+  }
+
+  private getGmsEmail(user: dbUser): string | undefined {
+    if (user.gmsAllowed && user.emailContact.gmsPublishEmail) {
+      return user.emailContact.email
+    }
+  }
+
+  private getGmsCountryCode(user: dbUser): string | undefined {
+    if (
+      user.gmsAllowed &&
+      (user.emailContact.gmsPublishPhone === GmsPublishPhoneType.GMS_PUBLISH_PHONE_COUNTRY ||
+        user.emailContact.gmsPublishPhone === GmsPublishPhoneType.GMS_PUBLISH_PHONE_FULL)
+    ) {
+      return user.emailContact.countryCode
+    }
+  }
+
+  private getGmsPhone(user: dbUser): string | undefined {
+    if (
+      user.gmsAllowed &&
+      user.emailContact.gmsPublishPhone === GmsPublishPhoneType.GMS_PUBLISH_PHONE_FULL
+    ) {
+      return user.emailContact.phone
+    }
+  }
+}
```

Comment:
I think the function names are a bit irritating.
For example:
```
GmsUser.getGmsAlias
```
what you would expect simply from the function name what you get?
Only together with the input it makes sense. On DbUser this names would make more sense, but this were the wrong place for this api specific logic.
Maybe you can extend the function names, for example:
```
GmsUser.getGmsAliasFromDbUser
```
Or use a separate Object for this Converter or Logic, I'm not sure what is better.
```ts
class DbUserConverter {
public constructor(private user: DbUser) {}
public getGmsAlias(): string | undefined {
    if (
      this.user.gmsAllowed &&
      this.user.alias &&
      this.user.gmsPublishName === GmsPublishNameType.GMS_PUBLISH_NAME_ALIAS_OR_INITALS
    ) {
      return this.user.alias
    }
  }
```

Reply:
I will agree with you if the functions are defined as public.
But here all they defined as privat functions and will in consequence be only invocable from inside the GmsUser-instance.
The GmsUser itself has no further public api than its own attributes.

As you write the private functions make sense in conjunction with the DbUser and this is only accessable in the constructor of the GmsUser.
The idea behind these functions is to hide the logic of the evaluation of the control settings and which value will be returned for the dedicated attribute like `getGmsAlias`, `getGmsFirstName` etc....

So i see no other or better context to shift them for a more efficient granuality.